docs(firebase_app_check): Demonstrate how to use App Check with Cloud Functions including with the Firebase Emulator. (#7571)
FirebaseExtended/flutterfire: docs/app-check/usage.mdx

Because this token allows access to your Firebase resources without a valid device, it is crucial that you keep it private.
Don't commit it to a public repository, and if a registered token is ever compromised, revoke it immediately in the
Firebase console.

> ## App Check for Functions with Firebase Local Emulator Suite

Reviewer: ```suggestion
## App Check for Functions
```

If you use [Firebase Local Emulator Suite](https://firebase.google.com/docs/emulator-suite) and have [App Check for Functions enabled](https://firebase.google.com/docs/app-check/cloud-functions), following the instructions you will check if app is verified app using the following code:

```

> if (context.app == undefined) {
>   // handle if function invocation comes from an unverified app, return and / or throw error
> }

Reviewer: ```suggestion
exports.yourCallableFunction = functions.https.onCall((data, context) => {
  // context.app will be undefined if the request doesn't include a valid
  // App Check token.
  // 
  if (context.app == undefined) {
    throw new functions.https.HttpsError(
        'failed-precondition',
        'The function must be called from an App Check verified app.')
  }

  // Your function logic follows.
});

```

```

> The above code will not allow your functions to run in Firebase Local Emulator Suite, because App Check token won't be added when the function is called. To solve this, change the above verification code to the following, so that App Check for functions is only enforced when not running in the Firebase Local Emulator Suite.

Reviewer: ```suggestion
You may also use the [Firebase Local Emulator Suite](https://firebase.google.com/docs/emulator-suite) to test your Cloud Function. You simply have to add another conditional check (i.e. `!process.env.FUNCTIONS_EMULATOR`) to check the veracity of the caller:
```

```

> if (context.app == undefined && !process.env.FUNCTIONS_EMULATOR) {
>   // handle if function invocation comes from an unverified app, return and / or throw error
> }

Reviewer: ```suggestion
exports.yourCallableFunction = functions.https.onCall((data, context) => {
  // context.app will be undefined if the request doesn't include a valid
  // App Check token.
  // 
  if (context.app == undefined && !process.env.FUNCTIONS_EMULATOR) {
    throw new functions.https.HttpsError(
        'failed-precondition',
        'The function must be called from an App Check verified app.')
  }
  // Your function logic follows.
});
```

Because this token allows access to your Firebase resources without a valid device, it is crucial that you keep it private.
Don't commit it to a public repository, and if a registered token is ever compromised, revoke it immediately in the
Firebase console.

## App Check for Functions

Ensure you have [App Check for Functions enabled](https://firebase.google.com/docs/app-check/cloud-functions), and check the context for the App Check token:

```
exports.yourCallableFunction = functions.https.onCall((data, context) => {
  // context.app will be undefined if the request doesn't include a valid
  // App Check token.

>   // 

Reviewer: ```suggestion
 
```

  if (context.app == undefined) {
    throw new functions.https.HttpsError(
        'failed-precondition',
        'The function must be called from an App Check verified app.')
  }

  // Your function logic follows.
});

```

> You may also use the [Firebase Local Emulator Suite](https://firebase.google.com/docs/emulator-suite) to test your Cloud Function. You simply have to add another conditional check (i.e. `!process.env.FUNCTIONS_EMULATOR`) to check the veracity of the caller:

Reviewer: ```suggestion
You may also use the [Firebase Local Emulator Suite](https://firebase.google.com/docs/emulator-suite) to test your Cloud Function. You simply have to add another conditional check (i.e. `!process.env.FUNCTIONS_EMULATOR`) to stop errors from throwing whilst testing:
```

Author: No problem!

```
exports.yourCallableFunction = functions.https.onCall((data, context) => {
  // context.app will be undefined if the request doesn't include a valid
  // App Check token.

>   // 

Reviewer: ```suggestion
 
```
